cpp_PAM_cross_service (#5083)
awsdocs/aws-doc-sdk-examples: cpp/example_code/cross-service/photo_asset_manager/README.md

* This code is not tested in every AWS Region. For more information, see [AWS Regional Services](https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services).

## Prerequisites

To run the code in this example, you need the following:

+ An AWS account.
+ Docker Desktop.
+ NodeJS 18+.
+ AWS SDK for C++. For SDK installation instructions, see [Get started with the AWS SDK for C++](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/getting-started.html).

## Create the resources

Follow the instructions in the
[README for the Photo Asset Manager application CDK](https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/applications/photo-asset-manager/cdk/README.md).
to use the AWS Cloud Development Kit (AWS CDK) or AWS Command Line Interface
(AWS CLI) to create and manage the resources used in this example. You must be running Docker in order to complete the steps for the AWS CDK.

## Build the code

The code is automatically built and uploaded when the resources are created.

The code is built using Docker and the provided [Dockerfile](Dockerfile).
The built code is located within the Docker container at `/pam_lambda/build/cpp_pam_lambdas.zip`.

> The code can be built and copied to the local file system with the following commands.

Reviewer: ```suggestion
To build and copy the code to the local file system, use the following commands.
```

```bash
docker build . -f Dockerfile -t cpp_pam_image
docker run --name pam_lambda cpp_pam_image
docker cp pam_lambda:/pam_lambda/build/cpp_pam_lambdas.zip .
```

### Application notes
* The maximum pixel size for analysis is 10000x10000. Larger images will not be analyzed.
* Some email clients do not support the long length download URLs. You might need to remove spaces from the URL before downloading.

## Delete the resources

To avoid charges, delete all the resources that you created for this tutorial.

> Follow the instructions in the [README for the Photo Asset Manager application cdk](https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/applications/photo-asset-manager/cdk/README.md)

Reviewer: ```suggestion
Follow the instructions in the [README for the Photo Asset Manager application CDK](https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/applications/photo-asset-manager/cdk/README.md)
```
